### PR TITLE
feat: Check validator subclasses are annotated with @GtfsValidator

### LINKED ISSUE #1262: Add unit-test to ensure all validator subclasses are annotated with @GtfsValidator
### Describe the problem

When implementing a new validation rule, I forgot to annotate my validator with the `@GtfsValidator` annotation.  This meant it wasn't picked up automatically by `ValidatorLoader` and not run as part of the default validation set, even though individual unit-tests for this validator ran successfully.

### Proposed solution

Add a unit-test that checks that any subclass of `SingleEntityValidator` or `FileValidator` is also annotated with `@GtfsValidator`.

### Alternatives you've considered

We could alternatively rewrite `ValidatorLoader` to ignore the `@GtfsValidator` annotation and instead scan for subclasses directly.  I'll admit I wasn't around for the original design choice, but I could see an argument for having a single annotation type across all validators, regardless of subclass, especially if the validator subclasses grows in the future.  Either way, given current staffing on the project, I'm in favor of keeping the status quo for now.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bdferris-v2**
Closes #1262

Add a unit-test that checks that any subclass of SingleEntityValidator or FileValidator is also annotated with @GtfsValidator.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues

### CODE REVIEW

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsValidatorAnnotationTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+              .scan()) {
+        for (Class<?> validatorBaseClass : validatorBaseClasses) {
+          for (ClassInfo classInfo : scanResult.getSubclasses(validatorBaseClass)) {
+            AnnotationInfo annotationInfo = classInfo.getAnnotationInfo(GtfsValidator.class);
+            assertWithMessage(
+                    classInfo.getName()
+                        + ": "
+                        + validatorBaseClass.getSimpleName()
+                        + " validator subclass should be annotated with @GtfsValidator")
+                .that(annotationInfo != null)
```
Comment (asvechnikov2) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsValidatorAnnotationTest.java:42]: you can probably use 
.that(annotationInfo).isNotNull();

**Review by asvechnikov2 — APPROVED** [review]
LGTM! Thanks!

File: main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsValidatorAnnotationTest.java (hunk 1 above)
Comment (bdferris-v2) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsValidatorAnnotationTest.java:42]: I originally wrote it this way, but the resulting error message from Truth isn't particular helpful / readable in the context of this check.  With the isTrue() check, we get something like:

```
SometValidator: FileValidator validator subclass should be annotated with @GtfsValidator
expected to be true
```

With the alternate syntax, we get:

```
SometValidator: FileValidator validator subclass should be annotated with @GtfsValidator
expected not to be: null
```
I think the former is more understandable than the latter.

Comment (asvechnikov2) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsValidatorAnnotationTest.java:42]: Thanks!

**Review by asvechnikov2 — APPROVED** [review]

**Pull request merged**